src/main/java/org/broadinstitute/hellbender/tools/exome/orientationbiasvariantfilter/OrientationBiasFilterer.java

```diff
@@ -294,11 +297,12 @@ static int calculateTotalNumToCut(final double fdrThreshold, final long unfilter
     public static Map<String, SortedMap<Genotype, VariantContext>> createSampleToGenotypeVariantContextSortedMap(final List<String> sampleNames, final Collection<VariantContext> variants) {
 
         // Sorts in reverse order (highest p_artifact goes first and will not allow anything to be equal
-        //  unless they share the same reference)
+        //  unless they share the same reference).  Unfortunately, we cannot check the actual reference, so we rely on
+        //  hashCode and the genotype rendered as a string to be universally unique.
         // Note the negative sign is to sort in reverse error.
         final Comparator<Genotype> genotypePArtifactComparator = Comparator
                 .comparingDouble((Genotype g) -> -OrientationBiasUtils.getGenotypeDouble(g, OrientationBiasFilterConstants.P_ARTIFACT_FIELD_NAME, 0.0))
-                .thenComparingInt(g -> g.hashCode());
+                .thenComparingInt(g -> g.hashCode()).thenComparing(Genotype::toString);
```

Comment:
Alternatively you could use a stateful comparator that tracks collision resolutions to convert the partial ordering by score into a TreeMap compatible total ordering.  http://gafter.blogspot.com/2007/03/compact-object-comparator.html